src/aof.c

```diff
@@ -1532,10 +1532,11 @@ int loadSingleAppendOnlyFile(char *filename) {
         }
 
         /* Command lookup */
+        sds err = NULL;
         cmd = lookupCommand(argv, argc);
-        if (!cmd) {
-            serverLog(LL_WARNING, "Unknown command '%s' reading the append only file %s", (char *)argv[0]->ptr,
-                      filename);
+        if ((!cmd && !commandCheckExistence(NULL, argv, argc, &err)) || (cmd && !commandCheckArity(cmd, argc, &err))) {
```

Comment:
Why not pass `fakeClient` to this function? It already has `fakeClient->argv` and `fakeClient->argc` so we don't need to add the new parameters.

```suggestion
        if ((!cmd && !commandCheckExistence(fakeClient, &err)) || (cmd && !commandCheckArity(cmd, argc, &err))) {
```

Reply:
I also thought about using it, but it requires assigning fakeClient->cmd, Ii haven't thought it through, i'll take a closer look to see if it has any impact.

Reply:
thanks, this look safe, i forgot freeClientArgv will nullify the cmd.